### PR TITLE
Xunit

### PR DESCRIPTION
**Pull request by pristinesource**
Converted to XUnit unit tests.  This will make things much easier for .net core.

### CODE REVIEW

File: ElasticDatabaseTools.sln [hunk 1]
```diff
@@ -23,6 +23,12 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "EntityFrameworkMultiTenant"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ElasticDapper", "Samples\Dapper\ElasticDapper.csproj", "{7B67F0A1-A5A9-477B-9B7E-FA2E16ED34DE}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Src", "Src", "{4621C430-14B7-4DD7-AAD3-D0D5AA7EF97C}"
```
Comment (pristinesource) [inline ElasticDatabaseTools.sln]: I restructured the files in the solution. Not required.

**Comment by pristinesource** [comment]
I thought I did sign a cla.  I can do it again though. :)

File: ElasticDatabaseTools.sln (hunk 1 above)
Comment (jaredmoo) [inline ElasticDatabaseTools.sln]: Ok, now that I open it in VS I see what this does. Cool.

**Comment by jaredmoo** [comment]
When I ran it in VS2015, all tests classes ran in parallel which caused many failures since all tests are targeting the same database. It would be preferable if the tests were changed to use different databases, but it would be an ok alternative if the tests could be configured to run sequentially (not in parallel). Is this possible?

**Comment by jaredmoo** [comment]
The change is good overall, I have only look at the first 25% so far but it looks like they are mostly boring mechanical changes so I expect the rest will be fine - will finish reviewing later. Thanks for getting this done! :)

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 2]
```diff
@@ -301,13 +306,13 @@ public void TestShardCommandBeginHandler()
                 }
             }
 
-            Assert.AreEqual(shardConnections.Count, passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
+            Assert.True(shardConnections.Count == passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 3]
```diff
@@ -546,7 +551,7 @@ public void TestShardCommandRetryConnectionReopen()
                         {
                             readerCount++;
                         }
-                        Assert.AreEqual(10, readerCount, "Expected 10 readers!");
+                        Assert.True(10 == readerCount, "Expected 10 readers!");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 4]
```diff
@@ -438,16 +443,16 @@ public void TestShardCommandRetryExhaustion()
...
                     {
-                        Assert.IsTrue(ex.InnerException is TimeoutException, "Expected TimeoutException!");
+                        Assert.True(ex.InnerException is TimeoutException, "Expected TimeoutException!");
                     }
 
                     // Validate that the connections for the faulted readers are closed
                     for (int i = 0; i < 5; i++)
                     {
-                        Assert.IsTrue(shardConnections[i].Item2.State == ConnectionState.Closed,
+                        Assert.True(shardConnections[i].Item2.State == ConnectionState.Closed,
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 5]
```diff
@@ -640,19 +645,19 @@ public void TestDataReaderReadException()
...
+                Assert.True(ex is InvalidOperationException, "Expected InvalidOperationException!");
             }
 
             // Validate that we didn't automatically move on to the next reader when we
             // hit an exception whilst reading the column and that 
             // an exception from a second Read() call is stored and the reader is closed
             multiShardDataReader.Read();
-            Assert.AreEqual(multiShardDataReader.MultiShardExceptions.Count, 1, "Expected exception to be recorded");
-            Assert.IsTrue(mockReader1.IsClosed, "Expected reader to be closed!");
+            Assert.True(multiShardDataReader.MultiShardExceptions.Count == 1, "Expected exception to be recorded");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 6]
```diff
@@ -136,13 +140,13 @@ public void TestShardCommandFaultHandler()
                 }
             }
 
-            Assert.AreEqual(shardConnections.Count, passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
+            Assert.True(shardConnections.Count == passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 7]
```diff
@@ -179,11 +183,11 @@ public void TestShardCommandCancellation()
 
                         cmd.ExecuteReader();
                         executeTask.Wait();
-                        Assert.Fail("We should always be throwing an exception.");
+                        AssertExtensions.Fail("We should always be throwing an exception.");
                     }
                     catch (Exception ex)
                     {
-                        Assert.IsTrue(ex is OperationCanceledException, "OperationCanceledException expected. Found {0}!", ex.ToString());
+                        Assert.True(ex is OperationCanceledException, String.Format("OperationCanceledException expected. Found {0}!", ex.ToString()));
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.IsType

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 8]
```diff
@@ -237,22 +241,21 @@ public void TestShardCommandCancellationHandler()
...
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.IsType

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 9]
```diff
@@ -237,22 +241,21 @@ public void TestShardCommandCancellationHandler()
...
                     }
                 }
             }
-            Assert.AreEqual(shardConnections.Count, passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
+            Assert.True(shardConnections.Count == passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 10]
```diff
@@ -328,15 +333,15 @@ public void TestShardCommandSucceedHandler()
...
+                        Assert.True(shardConnections.Select(x => x.Item1).Contains(eventArgs.ShardLocation), "The ShardLocation passed to the event handler does not exist in the set of passed in ShardLocations");
                         passedLocations[eventArgs.ShardLocation] = true;
                     });
                     CommandBehavior behavior = CommandBehavior.Default;
                     cmd.ExecuteReader(behavior);
                 }
             }
 
-            Assert.AreEqual(shardConnections.Count, passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
+            Assert.True(shardConnections.Count == passedLocations.Count, "Not every ShardLocation had its corresponding event handler invoked.");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 11]
```diff
@@ -438,16 +443,16 @@ public void TestShardCommandRetryExhaustion()
                     MultiShardDataReader rdr = cmd.ExecuteReader(CommandBehavior.Default);
 
                     // Validate the right exception is re-thrown
-                    Assert.IsTrue(rdr.MultiShardExceptions.Count == 5, "Expected MultiShardExceptions!");
+                    Assert.True(rdr.MultiShardExceptions.Count == 5, "Expected MultiShardExceptions!");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.Equal

File: Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs [hunk 12]
```diff
@@ -438,16 +443,16 @@ public void TestShardCommandRetryExhaustion()
...
                     foreach (MultiShardException ex in rdr.MultiShardExceptions)
                     {
-                        Assert.IsTrue(ex.InnerException is TimeoutException, "Expected TimeoutException!");
+                        Assert.True(ex.InnerException is TimeoutException, "Expected TimeoutException!");
```
Comment (jaredmoo) [inline Test/ElasticScale.Query.UnitTests/MultiShardUnitTests.cs]: Assert.IsType

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 13]
```diff
@@ -264,7 +163,7 @@ public void DateGetPointMappingsForRange()
                 while (mEnum.MoveNext())
                     count++;
             }
-            Assert.AreEqual(2, count);
+            Assert.True(2 == count);
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 14]
```diff
@@ -274,7 +173,7 @@ public void DateGetPointMappingsForRange()
...
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 15]
```diff
@@ -284,14 +183,14 @@ public void DateGetPointMappingsForRange()
                 while (mEnum.MoveNext())
                     count++;
             }
-            Assert.AreEqual(1, count);
+            Assert.True(1 == count);
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 16]
```diff
@@ -327,24 +226,24 @@ public void DateAddPointMappingDuplicate()
...
 
-            Assert.IsTrue(addFailed);
+            Assert.True(addFailed);
 
             PointMapping<DateTime> p2 = lsm.GetMappingForKey(val);
 
-            Assert.IsNotNull(p2);
-            Assert.AreEqual(0, countingCache.LookupMappingHitCount);
+            Assert.NotNull(p2);
+            Assert.True(0 == countingCache.LookupMappingHitCount);
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 17]
```diff
@@ -360,21 +259,21 @@ public void DateDeletePointMappingDefault()
...
 
             DateTime val = DateTime.Now;
             PointMapping<DateTime> p1 = lsm.CreatePointMapping(val, s);
 
             PointMapping<DateTime> p2 = lsm.GetMappingForKey(val);
 
-            Assert.IsNotNull(p2);
-            Assert.AreEqual(0, countingCache.LookupMappingHitCount);
+            Assert.NotNull(p2);
+            Assert.True(0 == countingCache.LookupMappingHitCount);
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs [hunk 18]
```diff
@@ -392,20 +291,20 @@ public void DateDeletePointMappingDefault()
...
-                Assert.AreEqual(ShardManagementErrorCode.MappingNotFoundForKey, sme.ErrorCode);
+                Assert.Equal(ShardManagementErrorCategory.ListShardMap, sme.ErrorCategory);
+                Assert.Equal(ShardManagementErrorCode.MappingNotFoundForKey, sme.ErrorCode);
                 lookupFailed = true;
             }
 
-            Assert.IsTrue(lookupFailed);
-            Assert.AreEqual(0, countingCache.LookupMappingMissCount);
+            Assert.True(lookupFailed);
+            Assert.True(0 == countingCache.LookupMappingMissCount);
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/DateTimeShardMapperTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj [hunk 19]
```diff
@@ -11,6 +12,8 @@
     <EnableFakes>true</EnableFakes>
     <IsCodedUITest>False</IsCodedUITest>
     <TestProjectType>UnitTest</TestProjectType>
+    <NuGetPackageImportStamp>
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj]: remove

File: Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj [hunk 20]
```diff
@@ -29,11 +32,7 @@
...
   <Choose>
-    <When Condition="'$(StandaloneBuild)' == 'true'">
-      <ItemGroup>
-        <Reference Include="FakesAssemblies\Microsoft.Azure.SqlDatabase.ElasticScale.Client.Fakes.dll" />
-      </ItemGroup>
-    </When>
+    <When Condition="'$(StandaloneBuild)' == 'true'" />
     <Otherwise>
       <ItemGroup>
         <Reference Include="Microsoft.Azure.SqlDatabase.ElasticScale.Client.Fakes, Version=$(VersionMajor).$(VersionMinor).0.0, Culture=neutral, PublicKeyToken=0ae41878053f6703, processorArchitecture=MSIL" />
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj]: Remove this - shouldn't be needed

File: Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj [hunk 21]
```diff
@@ -47,31 +46,64 @@
...
       <ItemGroup>
         <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />
       </ItemGroup>
     </When>
-    <Otherwise>
-      <ItemGroup>
-        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
-      </ItemGroup>
-    </Otherwise>
+    <Otherwise />
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj]: remove

File: Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj.user [hunk 22]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/ElasticScale.ShardManagement.UnitTests.csproj.user]: remove this file

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 23]
```diff
@@ -262,44 +190,44 @@ public void TestConsistencyDetectionAndViewingWithNoConflicts()
...
 
-            Assert.IsNotNull(r1);
+            Assert.NotNull(r1);
 
             RecoveryManager rm = new RecoveryManager(smm);
 
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 24]
```diff
@@ -262,44 +190,44 @@ public void TestConsistencyDetectionAndViewingWithNoConflicts()
...
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(0, kvps.Keys.Count, "An unexpected conflict was detected");
+                Assert.True(0 == kvps.Keys.Count, "An unexpected conflict was detected");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 25]
```diff
@@ -343,28 +271,28 @@ public void TestConsistencyDetectionAndViewingWithVersionOnlyConflict()
 
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 26]
```diff
@@ -343,28 +271,28 @@ public void TestConsistencyDetectionAndViewingWithVersionOnlyConflict()
...
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(1, kvps.Keys.Count, "An unexpected conflict was detected");
+                Assert.True(1 == kvps.Keys.Count, "An unexpected conflict was detected");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 27]
```diff
@@ -402,28 +330,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInLSM()
 
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 28]
```diff
@@ -402,28 +330,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInLSM()
...
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(2, kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(2 == kvps.Keys.Count, "The count of differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 29]
```diff
@@ -402,28 +330,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInLSM()
...
 
                 foreach (var kvp in kvps)
                 {
                     ShardRange range = kvp.Key;
                     MappingLocation mappingLocation = kvp.Value;
-                    Assert.AreEqual(1, (int)range.High.Value - (int)range.Low.Value, "The ranges reported differed from those expected.");
-                    Assert.AreEqual(MappingLocation.MappingInShardOnly, mappingLocation, "An unexpected difference between global and local shardmaps was detected. This is likely a false positive and implies a bug in the detection code.");
+                    Assert.True(1 == (int)range.High.Value - (int)range.Low.Value, "The ranges reported differed from those expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 30]
```diff
@@ -461,28 +389,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInGSM()
 
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 31]
```diff
@@ -461,28 +389,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInGSM()
...
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(2, kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(2 == kvps.Keys.Count, "The count of differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 32]
```diff
@@ -461,28 +389,28 @@ public void TestConsistencyDetectionAndViewingWithWiderRangeInGSM()
...
 
                 foreach (var kvp in kvps)
                 {
                     ShardRange range = kvp.Key;
                     MappingLocation mappingLocation = kvp.Value;
-                    Assert.AreEqual(1, (int)range.High.Value - (int)range.Low.Value, "The ranges reported differed from those expected.");
-                    Assert.AreEqual(MappingLocation.MappingInShardMapOnly, mappingLocation, "An unexpected difference between global and local shardmaps was detected. This is likely a false positive and implies a bug in the detection code.");
+                    Assert.True(1 == (int)range.High.Value - (int)range.Low.Value, "The ranges reported differed from those expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 33]
```diff
@@ -522,28 +450,28 @@ public void TestConsistencyDetectionAndViewingWithAdditionalRangeInLSM()
...
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(1, kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(1 == kvps.Keys.Count, "The count of differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 34]
```diff
@@ -522,28 +450,28 @@ public void TestConsistencyDetectionAndViewingWithAdditionalRangeInLSM()
 
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 35]
```diff
@@ -522,28 +450,28 @@ public void TestConsistencyDetectionAndViewingWithAdditionalRangeInLSM()
...
-                Assert.AreEqual(1, kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(1 == kvps.Keys.Count, "The count of differences does not match the expected.");
 
                 foreach (var kvp in kvps)
                 {
                     ShardRange range = kvp.Key;
                     MappingLocation mappingLocation = kvp.Value;
-                    Assert.AreEqual(20, (int)range.High.Value, "The range reported differed from that expected.");
-                    Assert.AreEqual(MappingLocation.MappingInShardOnly, mappingLocation, "An unexpected difference between global and local shardmaps was detected. This is likely a false positive and implies a bug in the detection code.");
+                    Assert.True(20 == (int)range.High.Value, "The range reported differed from that expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 36]
```diff
@@ -589,23 +517,23 @@ public void TestConsistencyDetectionOnListMapping()
...
-                Assert.AreEqual(1, kvps.Values.Where(l => l == MappingLocation.MappingInShardMapOnly).Count(), "The count of shardmap only differences does not match the expected.");
-                Assert.AreEqual(2, kvps.Values.Where(l => l == MappingLocation.MappingInShardOnly).Count(), "The count of shard only differences does not match the expected.");
-                Assert.AreEqual(1, kvps.Values.Where(l => l == MappingLocation.MappingInShardMapAndShard).Count(), "The count of shard only differences does not match the expected.");
+                Assert.True(4 == kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(1 == kvps.Values.Where(l => l == MappingLocation.MappingInShardMapOnly).Count(), "The count of shardmap only differences does not match the expected.");
+                Assert.True(2 == kvps.Values.Where(l => l == MappingLocation.MappingInShardOnly).Count(), "The count of shard only differences does not match the expected.");
+                Assert.True(1 == kvps.Values.Where(l => l == MappingLocation.MappingInShardMapAndShard).Count(), "The count of shard only differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Single in 3 above places

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 37]
```diff
@@ -589,23 +517,23 @@ public void TestConsistencyDetectionOnListMapping()
...
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
 
             foreach (RecoveryToken g in gs)
             {
                 var kvps = rm.GetMappingDifferences(g);
-                Assert.AreEqual(4, kvps.Keys.Count, "The count of differences does not match the expected.");
-                Assert.AreEqual(1, kvps.Values.Where(l => l == MappingLocation.MappingInShardMapOnly).Count(), "The count of shardmap only differences does not match the expected.");
-                Assert.AreEqual(2, kvps.Values.Where(l => l == MappingLocation.MappingInShardOnly).Count(), "The count of shard only differences does not match the expected.");
-                Assert.AreEqual(1, kvps.Values.Where(l => l == MappingLocation.MappingInShardMapAndShard).Count(), "The count of shard only differences does not match the expected.");
+                Assert.True(4 == kvps.Keys.Count, "The count of differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal in 2 above places

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 38]
```diff
@@ -732,7 +660,7 @@ public void TestConsistencyDetectionWithDivergence()
             {
                 var kvps = rm.GetMappingDifferences(g);
 
-                Assert.AreEqual(2, kvps.Keys.Count, "The count of differences does not match the expected.");
+                Assert.True(2 == kvps.Keys.Count, "The count of differences does not match the expected.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: Assert.Equal

File: Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs [hunk 39]
```diff
@@ -806,19 +734,19 @@ public void TestCopyGSMToLSM()
             // Briefly validate that there are, in fact, the two ranges of inconsistency we are expecting.
             IEnumerable<RecoveryToken> gs = rm.DetectMappingDifferences(sl);
 
-            Assert.AreEqual(1, gs.Count(), "The test environment was not expecting more than one local shardmap.");
+            Assert.True(1 == gs.Count(), "The test environment was not expecting more than one local shardmap.");
```
Comment (jaredmoo) [inline Test/ElasticScale.ShardManagement.UnitTests/RecoveryManagerTests.cs]: I will stop commenting about Assert.Equal, hopefully these can be found and replaced by regex :)

**Pull request closed without merging**